mountpoint-s3/src/data_cache/disk_data_cache.rs

```diff
@@ -276,30 +285,30 @@ mod tests {
         );
     }
 
-    #[test_case("hello"; "simple string")]
-    #[test_case("foo/bar/baz"; "with forward slashes")]
-    #[test_case("hello+world"; "with plus char")]
-    #[test_case("hello\\ world"; "backslash")]
-    #[test_case("hello=world"; "equals")]
-    #[test_case("look🎡emoji"; "emoji")]
-    fn test_get_path_for_key(s3_key: &str) {
+    #[test]
+    fn test_get_path_for_key() {
```

Comment:
minor point: maybe we could use `hash_cache_key` here and in `test_get_path_for_block` to build the expected path, and write a separate test specifically for `hash_cache_key`.

Reply:
sounds good! the thing I wanted to do was ensure we know when we change the encoding.